src/models/client.ts

```diff
@@ -209,28 +217,37 @@ export class ClientEnvironment {
     await this.network.clearBrowserCache();
   }
 
-  public async navigate(arg?: string | NavigateOptions): Promise<NavigateResult> {
+  private async convertToHash(responses: Network.ResponseReceivedParameters[], shouldAddBodies: boolean) {
+    const resHash = {} as ResponseHash;
+    const bodyExtractions = responses.map(async (response) => {
```

Comment:
I think map reduce might make this a little easier to read.

```
return responses.map(() => {
   ...
   return capturedResponse;
}).reduce((resHash: ResponseHash, capturedResponse) => {
   resHash[capturedResponse.url] = capturedResponse;
}, {});

Reply:
I believe this requires adding `url` to `CapturedResponse`, which might be good anyways.

Reply:
The issue here is that "getResponseBody()" is an async operation, but I don't want to just store that promise in resHash[url].body. I want to store the actual resolved body in resHash[url].body. This is because chrome clears away the response body buffer on the next navigation, so want to be storing hard copies of the bodies to be durable across navigates.

If this was not my requirement, I agree that a .map().reduce() would be cleaner. But as it stands, I haven't been able to figure out how to transform my async map into a .map().reduce() which functions the same way.

Let me know if there is a way to achieve the same result with .map().reduce().

Reply:
I had tried something like this before, but wasn't sure if using reduce() to fill out a hashtable was an acceptable usage. I'll rewrite it with map()/reduce(), but over multiple lines distinct to make it easier to digest.